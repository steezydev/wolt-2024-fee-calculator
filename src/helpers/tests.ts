const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
