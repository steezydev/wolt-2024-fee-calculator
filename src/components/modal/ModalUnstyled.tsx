import ButtonRound from '@/components/button/ButtonRound';
import CloseIcon from '@/components/icons/CloseIcon';
import B2 from '@/components/typography/B2';
import { classNames } from '@/helpers/classNames';
import useClickOutside from '@/hooks/useClickOutside';
import { ModalProps } from '@/types/props/ModalProps';

const ModalUnstyled = ({
  id,
  isOpen,
  onClose,
  title,
  children,
  className,
}: ModalProps) => {
  const ref = useClickOutside<HTMLDivElement>(() => onClose());

  if (!isOpen) {
    return null;
  }

  return (
    <div
      role='dialog'
      ref={ref}
      className={classNames('absolute z-50 flex flex-col gap-3', className)}
      aria-modal='true'
      aria-labelledby={`dialogTitle` + id}
    >
      <div className='border-b border-b-neutral-light-100 dark:border-b-neutral-dark-100 pb-1 flex justify-between'>
        <B2
          id={`dialogTitle` + id}
          className='text-dark dark:text-white font-bold'
        >
          {title}
        </B2>
        <ButtonRound
          id={`closeModalButton` + id}
          onClick={onClose}
          ariaLabel='Close modal'
        >
          <CloseIcon className='w-4 h-4' />
        </ButtonRound>
      </div>
      <div>{children}</div>
    </div>
  );
};

export default ModalUnstyled;
