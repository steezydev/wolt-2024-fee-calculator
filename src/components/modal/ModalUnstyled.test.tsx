import { modalEventTests, modalRenderTests } from './Modal.tests';
import ModalUnstyled from './ModalUnstyled';

modalRenderTests('ModalUnstyled', ModalUnstyled);
modalEventTests('ModalUnstyled', ModalUnstyled);
