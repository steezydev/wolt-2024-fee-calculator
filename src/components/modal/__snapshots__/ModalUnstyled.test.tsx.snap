// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalUnstyled when asked to render with children renders with children 1`] = `
<div
  aria-labelledby="dialogTitletestModal"
  aria-modal="true"
  className="absolute z-50 flex flex-col gap-3"
  role="dialog"
>
  <div
    className="border-b border-b-neutral-light-100 dark:border-b-neutral-dark-100 pb-1 flex justify-between"
  >
    <p
      className="text-lg text-dark dark:text-white font-bold"
      id="dialogTitletestModal"
    >
      Test modal
    </p>
    <button
      aria-label="Close modal"
      className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
      data-test-id="closeModalButtontestModal"
      id="closeModalButtontestModal"
      onClick={[MockFunction]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="w-4 h-4"
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 6 6 18"
        />
        <path
          d="m6 6 12 12"
        />
      </svg>
    </button>
  </div>
  <div>
    <div>
      Test modal content
    </div>
  </div>
</div>
`;

exports[`ModalUnstyled when asked to render without children renders without children 1`] = `
<div
  aria-labelledby="dialogTitletestModal"
  aria-modal="true"
  className="absolute z-50 flex flex-col gap-3"
  role="dialog"
>
  <div
    className="border-b border-b-neutral-light-100 dark:border-b-neutral-dark-100 pb-1 flex justify-between"
  >
    <p
      className="text-lg text-dark dark:text-white font-bold"
      id="dialogTitletestModal"
    >
      Test modal
    </p>
    <button
      aria-label="Close modal"
      className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
      data-test-id="closeModalButtontestModal"
      id="closeModalButtontestModal"
      onClick={[MockFunction]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="w-4 h-4"
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 6 6 18"
        />
        <path
          d="m6 6 12 12"
        />
      </svg>
    </button>
  </div>
  <div />
</div>
`;
